app/backend/src/clj/teet/thk/thk_import.clj

```diff
@@ -224,6 +232,112 @@
+     :project-est-start (first (sort phase-est-starts))
+     :project-est-end (last (sort phase-est-ends))
+
+     :proj-integration-id (:integration/id thk-project)
+     :proj-db-id (:db/id thk-project)
+
+     :project-exists? (some? (:db/id thk-project))
+     :project-has-owner? (and (some? (:db/id thk-project))
+                           (project-db/project-has-owner? db [:thk.project/id project-id]))
+     :construction-activity-db-id (get-project-construction-activity-db-id db rows)}))
+
+
+(defn- get-task-type [db task-eid]
+  (ffirst (d/q '[:find (pull ?e [:task/type])
+                 :where
+                 [(missing? $ ?e :meta/deleted?)]
+                 :in $ ?e] db task-eid)))
+
+(defn- get-activity-tasks-eids [db activity-eid]
+  (let [query-result
+        (d/q '[:find (pull ?e [:activity/tasks])
+               :in $ ?e] db activity-eid)]
+    (when
+      (some? query-result)
+      (:activity/tasks (ffirst query-result)))))
+
+(defn- get-existing-task-db-id
+  "Search for the existing task with the same type and start and end dates"
+  [db construction-activity-id task-type start-date end-date]
+  (first (reduce
+           (fn [acc task]
+             (if
+               (and
+                 (= (:db/ident (:task/type (get-task-type db (:db/id task)))) task-type)
+                 (= (:task/estimated-start-date (d/pull db '[*] (:db/id task))) start-date)
+                 (= (:task/estimated-end-date (d/pull db '[*] (:db/id task))) end-date))
+               (conj acc (:db/id task))
+               acc))
+           []
+           (get-activity-tasks-eids db construction-activity-id))))
+
+(defn- get-new-task-type [activity-data]
+  (let [activity-name (:activity/name activity-data)]
+    (case activity-name
+      :activity.name/owners-supervision :task.type/owners-supervision
+      :activity.name/road-safety-audit :task.type/road-safety-audit
+      (log/info "new task type unknown for activity " activity-name))))
+
+(defn add-task-from-thk
+  "Returns TX data for new task with given params for given construction activity-id"
+  [db thk-activity-task-data construction-activity-id ]
+  (let [start-date (:activity/estimated-start-date thk-activity-task-data)
+        end-date (:activity/estimated-end-date thk-activity-task-data)
+        thk-activity-status (:activity/status thk-activity-task-data)
+        task-type (get-new-task-type thk-activity-task-data)
+        existing-task-eid (get-existing-task-db-id db construction-activity-id task-type
+                               start-date end-date)
+        id-placeholder (if (some? existing-task-eid)
+                         existing-task-eid
+                         (str "NEW-TASK-" (name :task.group/construction) "-" task-type
+                           "-" (integration-id/unused-random-small-uuid db)))
+        task-tx-data (merge
+                       {:db/id id-placeholder
+                        :task/group (if (= :task.type/road-safety-audit task-type)
```

Comment:
should this be a case to enumerate all possibilities. Now if type is not road safety audit, it will be the other one. 
seems easy to miss if there are ever new types that need to be supported

Reply:
Refactored into the case.